test/e2e/upgrade_hypershift_operator_test.go

```diff
@@ -0,0 +1,247 @@
+		// otherwise skip it.
+		t.SkipNow()
+	}
+	var err error
+
+	var mgmtClient crclient.Client
+	var hostedCluster *hyperv1.HostedCluster
+	var hcpNameSpace string
+	var nodePoolsMap map[string]*hyperv1.NodePool
+	var machineDeploymentMap map[string]*v1beta1.MachineDeployment
+
+	hyperShiftOperatorLatestImage := globalOpts.HyperShiftOperatorLatestImage
+	hyperShiftOperatorInitialImage := globalOpts.HOInstallationOptions.HyperShiftOperatorInitialImage
+
+	g := gomega.NewWithT(t)
+
+	t.Parallel()
+	ctx, cancel := context.WithCancel(testContext)
+
+	defer cancel()
+
+	cleanup := install.NoOpCleanup
+	t.Cleanup(func() {
+		t.Log("Cleaning up HyperShift Operator")
+		err = cleanup(context.Background())
+		g.Expect(err).ToNot(gomega.HaveOccurred(), "Cleaning up initial HyperShift Operator shouldn't return errors")
+	})
+
+	clusterOpts := globalOpts.DefaultClusterOptions(t)
+
+	t.Log("Starting HyperShift Operator upgrade test")
+	zones := strings.Split(globalOpts.ConfigurableClusterOptions.Zone.String(), ",")
+	if len(zones) >= 3 {
+		t.Log("Sufficient zones available for InfrastructureAvailabilityPolicy HighlyAvailable")
+		clusterOpts.AWSPlatform.Zones = zones
+		clusterOpts.InfrastructureAvailabilityPolicy = string(hyperv1.HighlyAvailable)
+		clusterOpts.NodePoolReplicas = 1
+	}
+
+	clusterOpts.BeforeApply = func(o crclient.Object) {
+		switch obj := o.(type) {
+		case *hyperv1.HostedCluster:
+
+			// Add a label to identify the HostedCluster for upgrade tests in case they're leaked
+			// and need to be cleaned up.
+			if obj.Labels == nil {
+				obj.Labels = make(map[string]string)
+			}
+			obj.Labels[hostedClusterUpgradeTestLabel] = "true"
+		}
+	}
+
+	g.Expect(t.Run("Install initial HyperShift Operator", func(t *testing.T) {
+		t.Logf("Installing HyperShift Operator with image %s", hyperShiftOperatorInitialImage)
+		globalOpts.HyperShiftOperatorLatestImage = hyperShiftOperatorInitialImage
+		cleanup, err = e2eutil.InstallHyperShiftOperator(ctx, globalOpts)
+
+		if err != nil {
+			t.Fatal("Failed to install initial HyperShift Operator")
+		}
+
+	})).To(gomega.BeTrue(), "Installing initial HyperShift Operator should succeed")
+
+	client, err := e2eutil.GetClient()
+	g.Expect(err).ToNot(gomega.HaveOccurred(), "Getting kube client shouldn't return errors")
+
+	operatorImage, err := e2eutil.GetHyperShiftOperatorImage(ctx, client, globalOpts)
+	g.Expect(err).ToNot(gomega.HaveOccurred(), "Getting HyperShiftOperator image shouldn't return errors")
+	g.Expect(operatorImage).To(gomega.Equal(hyperShiftOperatorInitialImage))
+
+	t.Logf("Observed pre-upgrade HyperShift Operator image %q", operatorImage)
+
+	t.Log("Executing upgrade test")
+	e2eutil.NewHypershiftTest(t, ctx, func(t *testing.T, g gomega.Gomega, mc crclient.Client, hc *hyperv1.HostedCluster) {
+		t.Logf("HostedCluster %s created", crclient.ObjectKeyFromObject(hc))
+		// Sanity check the cluster by waiting for the nodes to report ready
+		t.Log("Waiting for nodes to report ready")
```

Comment:
we comment and log "waiting for nodes" but that's not what WaitForGuestClient does?